recipes/opengv/all/conanfile.py

```diff
@@ -0,0 +1,143 @@
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        tools.save(module_file, content)
+
+    def validate(self):
+        # Disable windows builds since they error out.
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows builds are not supported by this recipe.")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("eigen/3.4.0")
+        if self.options.with_python_bindings:
+            self.requires("pybind11/2.7.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["BUILD_PYTHON"] = self.options.with_python_bindings
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = "aarch64"
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        # Use conan's Eigen
+        old = """\
+            find_package(Eigen REQUIRED)
+            set(ADDITIONAL_INCLUDE_DIRS ${EIGEN_INCLUDE_DIRS} ${EIGEN_INCLUDE_DIR}/unsupported)"""
+
+        new = """\
+            find_package(Eigen3 REQUIRED)
+            set(ADDITIONAL_INCLUDE_DIRS ${Eigen3_INCLUDE_DIRS} ${Eigen3_INCLUDE_DIR}/unsupported)"""
+
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                            textwrap.dedent(old),
+                            textwrap.dedent(new)
+        )
+
+        # Use conan's pybind11
+        tools.replace_in_file(os.path.join(self._source_subfolder, "python", "CMakeLists.txt"),
+                            "add_subdirectory(pybind11)",
+                            "find_package(pybind11 REQUIRED)"
+        )
+
+        # Let conan handle fPIC / shared
+        old = """\
+            IF(MSVC)
+              set(BUILD_SHARED_LIBS OFF)"""
+
+        new = """\
+            IF(1)
+              #set(BUILD_SHARED_LIBS OFF)"""
+
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                            textwrap.dedent(old),
+                            textwrap.dedent(new)
+        )
+
```

Comment:
```suggestion
    def _patch_sources(self):
        # Use conan's Eigen
        old = """\
            find_package(Eigen REQUIRED)
            set(ADDITIONAL_INCLUDE_DIRS ${EIGEN_INCLUDE_DIRS} ${EIGEN_INCLUDE_DIR}/unsupported)"""

        new = """\
            find_package(Eigen3 REQUIRED)
            set(ADDITIONAL_INCLUDE_DIRS ${Eigen3_INCLUDE_DIRS} ${Eigen3_INCLUDE_DIR}/unsupported)"""

        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
                            textwrap.dedent(old),
                            textwrap.dedent(new)
        )

        # Use conan's pybind11
        tools.replace_in_file(os.path.join(self._source_subfolder, "python", "CMakeLists.txt"),
                            "add_subdirectory(pybind11)",
                            "find_package(pybind11 REQUIRED)"
        )

        # Let conan handle fPIC / shared
        old = """\
            IF(MSVC)
              set(BUILD_SHARED_LIBS OFF)"""

        new = """\
            IF(1)
              #set(BUILD_SHARED_LIBS OFF)"""

        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
                            textwrap.dedent(old),
                            textwrap.dedent(new)
        )

    def build(self):
        self._patch_sources()
```

Keep that patch separated. Also, it could be converted to a real patch file.

Reply:
I like using `replace_in_file()` instead of patches since they seem to work better when it comes to different versions of the libraries.
A new line in the file and the patch no longer works, whereas this is will (unless the new line touches this exact lines). 
If the conan policy is to use patches, I don't mind creating one.